File: tessoku/tessoku-book/src/bin/a03.rs
use proconio::input;

fn main() {
    input! {
      n: usize,
      k: usize,
      p: [usize; n],
      q: [usize; n]
    }

    for &r in &p {
        for &b in &q {
            if r + b == k {
                println!("Yes");
                return;
            }
        }
    }

    println!("No");
}

File: tessoku/tessoku-book/src/a03.py
# [A03 - Two Cards](https://atcoder.jp/contests/tessoku-book/tasks/tessoku_book_c)
def solve(_N: int, K: int, P: list[int], Q: list[int]):
    for p in P:
        for q in Q:
            if p + q == K:
                print("Yes")
                return
    print("No")


if __name__ == "__main__":
    [N, K] = map(int, input().split())
    P = list(map(int, input().split()))
    Q = list(map(int, input().split()))
    solve(N, K, P, Q)
